RxAndroid 3.0.1 NPE in JVM unit tests

Setup: JVM based unit tests with `RxAndroidPlugins.setMainThreadSchedulerHandler { TestScheduler() }`. Code under test containing e.g. `observeOn(AndroidSchedulers.mainThread())`.

Working fine on RxAndroid 3.0.0.

Updating to RxAndroid 3.0.1 failing with NPE:

```
java.lang.ExceptionInInitializerError
	at io.reactivex.rxjava3.android.schedulers.AndroidSchedulers.lambda$static$0(AndroidSchedulers.java:33)
	at io.reactivex.rxjava3.android.plugins.RxAndroidPlugins.callRequireNonNull(RxAndroidPlugins.java:85)
	at io.reactivex.rxjava3.android.plugins.RxAndroidPlugins.initMainThreadScheduler(RxAndroidPlugins.java:39)
	at io.reactivex.rxjava3.android.schedulers.AndroidSchedulers.<clinit>(AndroidSchedulers.java:33)
...
Caused by: java.lang.NullPointerException: looper == null
	at io.reactivex.rxjava3.android.schedulers.AndroidSchedulers.from(AndroidSchedulers.java:66)
	at io.reactivex.rxjava3.android.schedulers.AndroidSchedulers.from(AndroidSchedulers.java:54)
	at io.reactivex.rxjava3.android.schedulers.AndroidSchedulers$MainHolder.<clinit>(AndroidSchedulers.java:29)
```

Quick peek at commit history suggests this might be caused by #577

--- a/rxandroid/build.gradle
+++ b/rxandroid/build.gradle
@@ -12,6 +12,10 @@ android {
         sourceCompatibility rootProject.ext.sourceCompatibility
         targetCompatibility rootProject.ext.sourceCompatibility
     }
+
+    testOptions {
+        unitTests.returnDefaultValues = true
+    }
 }
 
 dependencies {

--- a/rxandroid/src/main/java/io/reactivex/rxjava3/android/schedulers/AndroidSchedulers.java
+++ b/rxandroid/src/main/java/io/reactivex/rxjava3/android/schedulers/AndroidSchedulers.java
@@ -25,8 +25,7 @@ import io.reactivex.rxjava3.core.Scheduler;
 public final class AndroidSchedulers {
 
     private static final class MainHolder {
-        static final Scheduler DEFAULT
-            = from(Looper.getMainLooper());
+        static final Scheduler DEFAULT = internalFrom(Looper.getMainLooper(), true);
     }
 
     private static final Scheduler MAIN_THREAD =
@@ -61,10 +60,13 @@ public final class AndroidSchedulers {
      *              locking. On API < 16 this value is ignored.
      * @see Message#setAsynchronous(boolean)
      */
-    @SuppressLint("NewApi") // Checking for an @hide API.
     public static Scheduler from(Looper looper, boolean async) {
         if (looper == null) throw new NullPointerException("looper == null");
+        return internalFrom(looper, async);
+    }
 
+    @SuppressLint("NewApi") // Checking for an @hide API.
+    private static Scheduler internalFrom(Looper looper, boolean async) {
         // Below code exists in androidx-core as well, but is left here rather than include an
         // entire extra dependency.
         // https://developer.android.com/reference/kotlin/androidx/core/os/MessageCompat?hl=en#setAsynchronous(android.os.Message,%20kotlin.Boolean)

